drivers/entropy/entropy_stm32.c

```diff
@@ -357,6 +357,18 @@ static int entropy_stm32_rng_init(const struct device *dev)
 	__ASSERT_NO_MSG(dev_cfg != NULL);
 
 #if CONFIG_SOC_SERIES_STM32L4X
+#if !defined(CONFIG_CLOCK_STM32_PLL_SRC_MSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSE)
+	/* If no PLLSRC clock, use other available 48 MHz clock source */
+#if defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI) && \
+	defined(CONFIG_CLOCK_STM32_MSI_RANGE) && \
+	CONFIG_CLOCK_STM32_MSI_RANGE == 11
+	LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_MSI);
+#else
+	LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_HSI48);
```

Comment:
Not all STM32L4 have HSI48 as RNG clock source. For example reference man RM0351 for stm32L4x6.

Reply:
Thank you for pointing this out.
In the rebased branch the HSI configuration (that is not expected to happen for STM32L47x/L48x devices) is checked first. In case no 48 MHz source clock is available an error is returned.

Reply:
Even the #endifs should be cleaner now.